mino/router/tree/example_test.go

```diff
@@ -0,0 +1,65 @@
+package tree
+
+import (
+	"fmt"
+
+	"go.dedis.ch/dela/mino"
+	"go.dedis.ch/dela/mino/minogrpc/session"
+)
+
+func ExampleRouter_New() {
+	router := NewRouter(session.AddressFactory{})
+
+	addrA := session.NewAddress("127.0.0.1:2000")
+	addrB := session.NewAddress("127.0.0.1:3000")
+
+	players := mino.NewAddresses(addrA, addrB)
+
+	table, err := router.New(players)
+	if err != nil {
+		panic("routing table failed: " + err.Error())
+	}
+
+	routes, voids := table.Forward(table.Make(addrA, []mino.Address{addrB}, []byte{}))
+	fmt.Println(voids)
+
+	for to := range routes {
+		fmt.Println(to)
+	}
+
+	// Output: map[]
+	// 127.0.0.1:3000
+}
+
+func ExampleTable_Prelude() {
+	routerA := NewRouter(session.AddressFactory{})
+
+	addrA := session.NewAddress("127.0.0.1:2000")
+	addrB := session.NewAddress("127.0.0.1:3000")
+
+	players := mino.NewAddresses(addrA, addrB)
+
+	table, err := routerA.New(players)
+	if err != nil {
+		panic("routing table failed: " + err.Error())
+	}
+
+	handshake := table.Prelude(addrB)
```

Comment:
Probably out of scope for this review:

I feel this example highlights that "Prelude" is not necessarily the best name for the interface.

It took me a while (and reading the rest of the documentation) to understand the intention of this example, as "Prelude" really doesn't say *what* this is doing, but only *when* it should be done.

Reply:
Oh I would definitely use a different name, I just couldn't find one !